### ISSUE TITLE
Bar chart pick the wrong data in dataset.

### ISSUE THREAD

**Issue by icetimidus** [apache/incubator-echarts#11322]
### Version
4.2.1

### Reproduction link
[https://gallery.echartsjs.com/editor.html?c=xZQjtEQlY7&v=1](https://gallery.echartsjs.com/editor.html?c=xZQjtEQlY7&v=1)

### Steps to reproduce
See the link above.

### What is expected?
The data in chart is correct.

### What is actually happening?
The data of "公司业务系统" in bar chart is wrong. For example, in row three, the data of "公司业务系统" should be 1332.5333333333333. But it is 1666775.3333333333, as same as "ECM".

---
Duplicate of #11226 . Because of long time for waiting, I had to create a new issue.

**Comment by Ovilia** [comment]
Do you mind creating a minimum demo for this? To move the unnecessary parts and provide a demo with minimum series and minimum data.

**Comment by icetimidus** [comment]
@Ovilia 
Here is a minimum option.

option = {
    "dataset": {
        "source": [{
                "timestamp": 1568191020000,
                "ECM": 26311.466666666667,
            },
            {
                "timestamp": 1568191140000,
                "ECM": 1666775.3333333333,
                "GSWY": 1332.5333333333333,
            },
        ]
    },
    "xAxis": {
        "type": "time",
        "axisLabel": {
            "interval": 0
        }
    },
    "yAxis": {
        "type": "value",
    },
    "series": [{
            "type": "bar",
            "name": "ECM",
            "encode": {
                "x": "timestamp",
                "y": "ECM"
            },
            "stack": "one",
            "itemStyle": {
                "normal": {
                    "color": "#2ec7c9"
                }
            }
        },
        {
            "type": "bar",
            "name": "GSWY",
            "encode": {
                "x": "timestamp",
                "y": "GSWY"
            },
            "stack": "one",
            "itemStyle": {
                "normal": {
                    "color": "#97b552"
                }
            }
        }
    ],
    "tooltip": {}
}

**Comment by Ovilia** [comment]
I think there is a missing entry of `"GSWY"` in the first item of the dataset. Is this a mistake?
But ECharts should handle this kind of input and this could be treated as a bug.

**Comment by icetimidus** [comment]
@Ovilia 
No, the data will missing when it be zero. I think that's the problem is.

**Comment by Ovilia** [comment]
You should probably check the data before passing to ECharts for now.

**Comment by icetimidus** [comment]
Ok, I hope this bug could be fixed soon. Please close this issue if it should be.

**Comment by hoangv** [comment]
@icetimidus I spent some time studying this bug and reading [the documentation](https://www.echartsjs.com/en/option.html#dataset.source).  It appears the library allows the option to specify the dimensions for dataset.source.  When not explicitly specified, it makes a best-effort to detect dimensions on its own.  In your minimum example, it uses the first data point, which unfortunately happens to be the one missing GSWY dimension.  It seems resonable to first check the data and dimensions beforehand.  If you aren't able to check your data and dimensions beforehand, there are a couple options. 

1) In the minimal example, reverse the order of dataset.source so the first data point contains all the dimensions you want to plot.  For example:

```
    "dataset": {
        "source": [
            {
                "timestamp": 1568191140000,
                "ECM": 1666775.3333333333,
                "GSWY": 1332.5333333333333,
            },
            {
                "timestamp": 1568191020000,
                "ECM": 26311.466666666667,
            },
        ]
```
2) Explicitly specify all your dimensions in option.series

```
    "series": [{
            "type": "bar",
            "name": "ECM",
            "dimensions": ['timestamp','ECM','GSWY'],
            "encode": {
                "x": "timestamp",
                "y": "ECM"
            },
            "stack": "one",
            "itemStyle": {
                "normal": {
                    "color": "#2ec7c9"
                }
            }
        },
        {
            "type": "bar",
            "name": "GSWY",
            "dimensions": ['timestamp','ECM','GSWY'],
            "encode": {
                "x": "timestamp",
                "y": "GSWY"
            },
            "stack": "one",
            "itemStyle": {
                "normal": {
                    "color": "#97b552"
                }
            }
        }
    ],
```

Hope this helps.  Cc. @Ovilia
